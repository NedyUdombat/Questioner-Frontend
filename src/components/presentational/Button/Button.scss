@import '../../../assets/sass/index';

.btn {
  align-items: center;
  background-color: $primary-grey;
  border-radius: .3125rem;
  cursor: pointer;
  display: flex;
  font: 1rem $raleway;
  height: 3.125rem;
  justify-content: center;
  letter-spacing: .1em;
  outline: none;
  width: 100%;
}

.btn-dark {
  background-color: $primary-black;
  border: .0625rem solid $primary-black;
  color: $primary-white;
}

.btn-dark:hover {
  background-color: $primary-white;
  border: .0625rem solid $primary-black;
  color: $primary-black;
}

.btn-dark-alt {
  background-color: $primary-black;
  border: .0625rem solid $primary-white;
  color: $primary-white;
}

.btn-dark-alt:hover {
  background-color: $primary-white;
  border: .0625rem solid $primary-black;
  color: $primary-black;
}

.btn-white {
  background-color: $primary-white;
  border: .0625rem solid $primary-black;
  color: $primary-black;
}

.btn-white:hover {
  background-color: $primary-black;
  border: 1px solid $primary-black;
  color: $primary-white;
}

.btn-white-alt-hover {
  background-color: $primary-white;
  border: .0625rem solid $primary-black;
  color: $primary-black;
}

.btn-white-alt-hover:hover {
  background-color: $primary-black;
  border: 1px solid $primary-white;
  color: $primary-white;
}

.btn-transparent {
  background-color: transparent;
  color: $primary-white;
}

.btn-transparent:hover {
  background-color: transparent;
  border-bottom: .0625rem solid $primary-white;
  border-radius: 0;
  color: $primary-white;
  transition: .1s ease-in;
}

.btn-danger {
  background-color: $primary-red;
  border: .0625rem solid $primary-red;
  color: $primary-white;
}

.btn-danger:hover {
  background-color: $primary-white;
  border: .0625rem solid $primary-red;
  color: $primary-red;
}
