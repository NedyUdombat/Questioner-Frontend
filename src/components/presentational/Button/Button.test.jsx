import React from 'react';
import { mount } from 'enzyme';

import Button from './Button';

const clickTest = jest.fn();

describe('Button', () => {
  it('should render correctly', () => {
    const component = mount(
      <Button className="btn" type="button" value="login" onClick={clickTest()}>
        hey
      </Button>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const component = mount(
      <Button className="btn" type="button" onClick={clickTest()}>
        hey
      </Button>,
    );
    expect(component).toMatchSnapshot();
  });
});
