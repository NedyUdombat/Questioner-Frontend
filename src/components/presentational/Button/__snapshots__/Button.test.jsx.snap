// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<Button
  className="btn"
  type="button"
  value="login"
>
  <button
    className="btn btn"
    type="button"
  >
    login
  </button>
</Button>
`;

exports[`Button should render correctly 2`] = `
<Button
  className="btn"
  type="button"
>
  <button
    className="btn btn"
    type="button"
  >
    hey
  </button>
</Button>
`;
